File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        if(nums.Length == 0) return 0;
        if(nums.Length == 1) return nums[0];
        
        int sum=0, ret=int.MinValue;
        
         foreach(int n in nums)
         {
            sum += n;
            ret = Math.Max(sum, ret);
            if(sum < 0) sum = 0;
        }
        
        return ret;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.ts
function maxSubArray(nums: number[]): number {
    if(nums.length == 0) return 0
    if(nums.length == 1) return nums[0];
    
    let sum: number = 0;
    let ret: number = nums[0];
    
    for(const n of nums){
        sum+=n;
        if(ret < sum) ret = sum;
        if(sum < 0) sum = 0;
    }
    
    return ret;
};